File: core/decision/evidence.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(slots=True)
class Evidence:
    """
    One individual piece of evidence that contributed
    to a PredatorAI decision.
    """

    name: str

    source: str

    description: str

    impact: float

    confidence: float
    

File: frontend/src/reasoning/Evidence.ts
import type { EvidenceSource } from "./EvidenceSource";
import type { EvidenceType } from "./EvidenceType";

export interface Evidence {
    id: string;
    type: EvidenceType;
    source: EvidenceSource;
    title: string;
    description: string;
    confidence: number;
    timestamp: string;
    weight: number;
}
